browser/brave_content_browser_client.cc

```diff
@@ -345,3 +345,23 @@ GURL BraveContentBrowserClient::GetEffectiveURL(
   return url;
 #endif
 }
+
+void BraveContentBrowserClient::GetStoragePartitionConfigForSite(
+    content::BrowserContext* browser_context,
+    const GURL& site,
+    bool can_be_default,
+    std::string* partition_domain,
+    std::string* partition_name,
+    bool* in_memory)
+{
+  ChromeContentBrowserClient::GetStoragePartitionConfigForSite(
+      browser_context, site, can_be_default, partition_domain, partition_name,
+      in_memory);
+  if (browser_context->IsTorProfile()) {
+    if (!site.is_empty()) {
+      *partition_domain = site.host();
+      partition_name->clear();
```

Comment:
I think it is safer to set a partition name here like "tor" to prevent collision that stated here
https://chromium.googlesource.com/chromium/src/+/242ee8f02df8fc893dba3682b43d45e91db57580/content/browser/storage_partition_impl_map.cc#351